### ISSUE TITLE
loss or weight norm is nan. Training Stopped!

### ISSUE THREAD

**Issue by horizonheart** [holyseven/PSPNet-TF-Reproduce#15]
< Finetuning Process: not import resnet_v1_101/psp/pool1/BatchNorm/gamma:0 >
< Finetuning Process: not import resnet_v1_101/block4/unit_4/weights:0 >
< Finetuning Process: not import resnet_v1_101/block4/unit_4/BatchNorm/beta:0 >
< Finetuning Process: not import resnet_v1_101/block4/unit_4/BatchNorm/gamma:0 >
< Finetuning Process: not import resnet_v1_101/logits/weights:0 >
< Finetuning Process: not import resnet_v1_101/logits/biases:0 >
< Succesfully loaded fine-tune model from /nfs/private/PSPNet-TF-Reproduce-master/z_pretrained_weights/resnet_v1_101.c
kpt. >

< training process begins >

loss or weight norm is nan. Training Stopped!

**Comment by holyseven** [comment]
Try to use a larger batch size or reduce the learning rate / weight decay.

**Comment by horizonheart** [comment]
i found the loss is not nan ,but the wd is nan .is there any relationship with learning rate or weight decay and i used the default value.

**Comment by holyseven** [comment]
OK. Could you tell me the configuration you were using, train_image_size, batch_size, gpu number,  learning rate, weight decay rate(s) and the database.

**Comment by horizonheart** [comment]
batch and gpu is 1,other is deault.the dataset is Mapillary. the wd will become nan at the second itertion when i used tf.print.Do you have run your code successfully?

**Comment by holyseven** [comment]
I didn't try it on Mapillary. But batch size * gpu_num should be larger than 16 (suggested by the PSPNet paper)

**Comment by horizonheart** [comment]
> I didn't try it on Mapillary. But batch size * gpu_num should be larger than 16 (suggested by the PSPNet paper)

But the problem now is that the parameters of the model appear nan, this has nothing to do with the size of the batch size.

**Comment by holyseven** [comment]
One example in a batch will cause the training unstable (probably due to the batch normalization layer), leading the parameters exploding, i.e. NAN.

**Comment by horizonheart** [comment]
I used four gpus, and the batchsize is set to 4 ，but there is still NAN. Are you sure your program can run?

**Comment by holyseven** [comment]
Sorry, I didn't try Mapillary dataset with my code. Based on this situation, I cannot tell the reason of NAN. Try to use one of ADE20K, Pascal VOC or Cityscapes databases, and see if the NAN is still there.
